File: DSA/Arshdeep-Kaur/Day13_Question2.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindromeHelper(const string &str, int start, int end) {
    if (start >= end) {
        return true;
    }
    if (str[start] != str[end]) {
        return false;
    }
    return isPalindromeHelper(str, start + 1, end - 1);
}

string isPalindrome(int number) {
    string str = to_string(number);
    
    if (isPalindromeHelper(str, 0, str.length() - 1)) {
        return "YES";
    } else {
        return "NO";
    }
}

int main() {
    int number = 121;
    cout << isPalindrome(number) << endl;

    return 0;
}

File: DSA/Vikash_kumar/Day_13_Question_2.py
#Day13-Question2-
def is_palindrome_recursive(s, start, end):
    if start >= end:
        return True
    if s[start] != s[end]:
        return False

    return is_palindrome_recursive(s, start + 1, end - 1)

def is_palindrome(number):
    s = str(number)

    if is_palindrome_recursive(s, 0, len(s) - 1):
        return "YES"
    else:
        return "NO"

number = int(input("Enter the number: "))
print(is_palindrome(number)) 
